### ISSUE TITLE
New Rule: no-empty-headings

### ISSUE THREAD

**Issue by minseonkkim** [eslint/markdown#742]
### Rule details

Disallow headings that have no content (e.g. `##`, `### `, `### ###`).

### What type of rule is this?

Warns about a potential problem

### Example code

```markdown
<!-- eslint markdown/no-empty-headings: "error" -->

# Title

##

Some text.

### 

More text.

## ##
```

### Prior Art

```markdown
I couldn't find an equivalent rule in Markdown linters, but the same check exists in HTML accessibility tooling:

- axe-core `empty-heading` ("Headings should not be empty"): https://dequeuniversity.com/rules/axe/4.13/empty-heading
- W3C ACT rule ffd0e9 "Heading has non-empty accessible name": https://www.w3.org/WAI/standards-guidelines/act/rules/ffd0e9/

Since every empty ATX heading renders as an empty `<hN></hN>` element, it causes exactly the problem these rules detect.
```

### Participation

- [x] I am willing to submit a pull request to implement this rule.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

### Additional comments

> Disclosure: I'm a participant of the open source contribution program [OSSCA](https://github.com/eslint-ossca).

**Comment by lumirlumir** [comment]
> Disclosure: I'm a participant of the open source contribution program [OSSCA](https://github.com/eslint-ossca): confirmed.

I’m personally in favor of this rule, so +1 from me. We already have `no-empty-definitions`, `no-empty-images`, and `no-empty-links` rules that behave similarly, and the CommonMark specification allows empty ATX or closed ATX headings.

Since this is a feature request, I’m leaving it open for others to review as well.

**Comment by DMartens** [comment]
I would be -1 on adding a new rule specifically only for checking for an empty heading,
as there other "blocks" which can be empty and we should not add a rule for each one.
Rather I think we should consolidate everything into a single rule `no-empty-block`, which additionally checks for empty code fences, blockquotes, list items and frontmatters.
In a future major version we could merge the existing rules `no-empty-definitions/images/links`.
